src/istio/dynamic_encoding/util.cc

```diff
@@ -0,0 +1,193 @@
+/* Copyright 2018 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "util.h"
+#include "absl/strings/str_format.h"
+#include "google/protobuf/descriptor.h"
+#include "google/protobuf/descriptor.pb.h"
+
+namespace istio {
+namespace dynamic_encoding {
+using ::google::protobuf::Descriptor;
+using ::google::protobuf::EnumDescriptor;
+using ::google::protobuf::FieldDescriptor;
+using ::google::protobuf::FileDescriptorSet;
+using ::google::protobuf::util::Status;
+using ::google::protobuf::util::StatusOr;
+using ::google::protobuf::util::error::INTERNAL;
+using int32 = std::int32_t;
+using uint32 = std::uint32_t;
+using int64 = std::int64_t;
+using uint64 = std::uint64_t;
+
+namespace {
+
+Status getFieldEncodingError(const FieldDescriptor* fieldDescriptor){
+  const std::string err_msg = absl::StrFormat(
+      "unable to encode: %s for field %s", fieldDescriptor->type_name(),
+      fieldDescriptor->name());
+  return Status(INTERNAL, err_msg);
+}
+
+}
+
+Status EncodeStaticField(
+    absl::any* value, MessageEncoder* messageEncoder,
+    const FieldDescriptor* fieldDescriptor,
+    const EnumDescriptor* enumDescriptor, int index) {
+  switch (fieldDescriptor->cpp_type()) {
+    case FieldDescriptor::CPPTYPE_STRING : {
+      std::string* str = absl::any_cast<std::string>(value);
+      if (str == nullptr) {
+        return getFieldEncodingError(fieldDescriptor);
+      }
+      if (fieldDescriptor->is_repeated()) {
```

Comment:
Can we use macros here?
This is a lot of repetitive code.

Reply:
Will look into that..

Reply:
Fixed